platform/default/include/mbgl/map/map_snapshotter.hpp

```diff
@@ -10,29 +9,39 @@
 #include <string>
 #include <vector>
 #include <functional>
-
 namespace mbgl {
 
-template<class> class ActorRef;
 struct CameraOptions;
-class FileSource;
-class Size;
 class LatLngBounds;
 class ResourceOptions;
 
 namespace style {
 class Style;
 } // namespace style
 
+class MapSnapshotterObserver {
+public:
+    virtual ~MapSnapshotterObserver() = default;
+
+    static MapSnapshotterObserver& nullObserver() {
+        static MapSnapshotterObserver mapSnapshotterObserver;
```

Comment:
this needs to be put to a .cpp file, otherwise it's copied for every compilation unit.

Reply:
moved to cpp